pkg/cmd/label/edit.go

```diff
@@ -0,0 +1,104 @@
+package label
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type editOptions struct {
+	BaseRepo   func() (ghrepo.Interface, error)
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+
+	Color       string
+	Description string
+	Name        string
+	NewName     string
+}
+
+func newCmdEdit(f *cmdutil.Factory, runF func(*editOptions) error) *cobra.Command {
+	opts := editOptions{
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "edit <name>",
+		Short: "Edit a label",
+		Long: heredoc.Docf(`
+			Update a label on GitHub.
+
+			A label can be renamed using the %[1]s--name%[1]s flag.
+
+			The label color needs to be 6 character hex value.
+		`, "`"),
+		Example: heredoc.Doc(`
+			# update the color of the bug label
+			$ gh label edit bug --color FF0000
+
+			# rename and edit the description of the bug label
+			$ gh label edit bug --name big-bug --description "Bigger than normal bug"
+	  `),
+		Args: cmdutil.ExactArgs(1, "cannot update label: name argument required"),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts.BaseRepo = f.BaseRepo
+			opts.Name = args[0]
+			opts.Color = strings.TrimPrefix(opts.Color, "#")
+			if opts.Description == "" &&
+				opts.Color == "" &&
+				opts.NewName == "" {
+				return cmdutil.FlagErrorf("specify at least one of `--color`, `--description`, or `--name`")
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return editRun(&opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Description, "description", "d", "", "Description of the label")
+	cmd.Flags().StringVarP(&opts.Color, "color", "c", "", "Color of the label")
+	cmd.Flags().StringVarP(&opts.NewName, "name", "n", "", "Name of the label")
+
+	return cmd
+}
+
+func editRun(opts *editOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	baseRepo, err := opts.BaseRepo()
+	if err != nil {
+		return err
+	}
+
+	opts.IO.StartProgressIndicator()
+	err = updateLabel(apiClient, baseRepo, opts)
+	opts.IO.StopProgressIndicator()
+	if err != nil {
+		if errors.Is(err, errLabelAlreadyExists) {
+			return fmt.Errorf("label with name %q already exists", opts.NewName)
+		}
+		return err
+	}
+
+	if opts.IO.IsStdoutTTY() {
+		cs := opts.IO.ColorScheme()
+		successMsg := fmt.Sprintf("%s Label %q updated in %s\n", cs.SuccessIcon(), opts.Name, ghrepo.FullName(baseRepo))
```

Comment:
@samcoe I noticed you changed this to use `ghrepo.FullName()` here and in another PR. Was there some reason not to rely on `ghrepo.Interface.String()` (i.e. just pass `baseRepo` like I was) given our [thread here](https://github.com/cli/cli/pull/5452#discussion_r854793489)? Should I remove `String()`?

Reply:
Yeah we may have been overzealous with adding `String()`. The underlying type of `baseRepo` here is actually a `context.Remote` which satisfies the `ghrepo.Interface`. Unfortunately, `context.Remote` already has a `String()` function by way of `git.Remote`, so `baseRepo` was outputting the `git.Remote` name rather than the repo name. I could have went ahead and fixed that behavior but I was unsure of where in the code we might be relying on that behavior so I went with the safer fix of using `ghrepo.FullName`.

Reply:
In case you do want it removed: #5536